Token.js

```diff
@@ -3543,6 +3547,16 @@ function paste_selected_tokens() {
 		}
 		window.TOKEN_OBJECTS[newId].selected = true;
 		window.TOKEN_OBJECTS[newId].place_sync_persist();
+		if (id in window.JOURNAL.notes) {
+			window.JOURNAL.notes[newId] = window.JOURNAL.notes[id];
+			let copiedNote = window.JOURNAL.notes[newId];
```

Comment:
I think you need to make a clone/copy of the note here. When I copy and paste tokens and then edit the note - refreshing changes all their notes to be the same. I think this is cause you're just referencing the original (not 100% sure though)

This section should also probably be moved above the place of the token so that the note appears when pasted instead of on the next token update.

Example video
https://i.imgur.com/U7kkCRj.mp4

Reply:
looks good otherwise and good catch on the notes not being deleted on token delete.

Reply:
Ah yeah, nice catch. I was thinking that was going to happen, but could get it to trigger. Thanks for catching!